drivers/clock_control/clock_control_nrf.c

```diff
@@ -79,6 +79,9 @@ struct nrf_clock_control_config {
 static atomic_t hfclk_users;
 static uint64_t hf_start_tstamp;
 static uint64_t hf_stop_tstamp;
+#if CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH
+static bool keep_hfxo; /* If true, HFXO is running only to synthesize low frequency clock. */
```

Comment:
i'm afraid that this will impact the flow of onoff manager which manages all hfxo requests (there is also BT that can request HFXO in a special way). It would be better to use `onoff_manager` for that.
Here create `struct onoff_client lfsyth_cli`.

Then in a `lfclk_start()` do following:
```
// init notify client to not generate any callback.
sys_notify_init_spinwait(&data->lfsynth_cli.notify);
// request hfxo
(void)onoff_request(z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF), &data->lfsyth_cli);
```

in lfclk_stop()
```
(void)onoff_cancel_or_release(z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF), &data->lfsyth_cli);
```

onoff manager takes care of any other asynchronous requests or releases of hfxo that may come when lfsynth is requesting it.